lib/test.js

```diff
@@ -699,10 +724,19 @@ class Test {
 		};
 	}
 
-	finishPromised() {
```

Comment:
Looking at how this is called, I think we can get rid of this function entirely and just make `finish()` `async`. But that's an unrelated refactor.

Reply:
Yeah, I was considering it but didn't have enough know-how to make that call. Also, as far as I could tell, up until now `finish()` didn't really do anything `async`-ish.

When you say unrelated, do you want me to do that in a separate PR?

Reply:
I'm gonna go ahead and do this in a separate PR as there are so many lines affected, to not clutter this PR unnecessarily.

Reply:
Done: #2463